library/Vanilla/Formatting/Html/DomUtils.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * @author Dani M. <dani,m@vanillaforums.com>
+ * @author Isis Graziatto <isis.g@vanillaforums.com>
+ * @copyright 2009-2020 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla\Formatting\Html;
+
+use DOMDocument;
+
+/**
+ * Class for stripping images and truncating text from a Dom.
+ */
+final class DomUtils {
+
+    /** @var array */
+    const EMBED_CLASSES = ['js-embed', 'embedResponsive', 'embedExternal', 'embedImage', 'VideoWrap', 'iframe'];
+
+    /**
+     * Remove embeds from the dom.
+     *
+     * @param DOMDocument $dom
+     * @param array $embedClasses
+     */
+    public static function stripEmbeds(DOMDocument $dom, array $embedClasses = self::EMBED_CLASSES): void {
+        $xpath = new \DomXPath($dom);
+        foreach ($embedClasses as $key => $value) {
+            $xpathQuery = $xpath->query(".//*[contains(@class, '$embedClasses[$key]')]");
+            $xpathDivQuery = $xpath->query("//div[@data-embedjson]");
+            $dataClassItem = $xpathQuery->item(0);
+            $dataDivItem = $xpathDivQuery->item(0);
+            if ($dataClassItem) {
+                $dataClassItem->parentNode->removeChild($dataClassItem);
+            } elseif ($dataDivItem) {
+                $dataDivItem->parentNode->removeChild($dataDivItem);
```

Comment:
If you install the code coverage chrome extensions, you'll see that this line of code wasn't hit by a test. Can you conceive of a test that hits this?

Reply:
Very useful, that was the case of uploading a file in rich-editor which we haven't added a test case for yet. Isis will be adding that.